File: SITA/SITA2/models/Thesis.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const Thesis = sequelize.define('Thesis', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  student_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'users',
      key: 'id'
    }
  },
  title: {
    type: DataTypes.STRING(255),
    allowNull: false,
    validate: {
      notEmpty: true
    }
  },
  research_object: {
    type: DataTypes.TEXT,
    allowNull: false,
    validate: {
      notEmpty: true
    }
  },
  methodology: {
    type: DataTypes.TEXT,
    allowNull: false,
    validate: {
      notEmpty: true
    }
  },
  attachment_file: {
    type: DataTypes.STRING(255),
    allowNull: false
  },
  status: {
    type: DataTypes.ENUM('pending', 'approved', 'rejected'),
    defaultValue: 'pending'
  },
  rejection_reason: {
    type: DataTypes.TEXT,
    allowNull: true
  }
}, {
  tableName: 'thesis_submissions',
  timestamps: true,
  underscored: true
});

module.exports = Thesis;

File: app/src/main/java/com/example/sitaa/model/Thesis.kt
package com.example.sitaa.model

import com.google.gson.annotations.SerializedName
import java.util.Date

data class Thesis(
    val id: Int,
    @SerializedName("student_id")
    val studentId: Int,
    val title: String,
    @SerializedName("research_object")
    val researchObject: String,
    val methodology: String,
    @SerializedName("attachment_file")
    val attachmentFile: String,
    val status: String,
    @SerializedName("rejection_reason")
    val rejectionReason: String? = null,
    @SerializedName("created_at")
    val createdAt: Date,
    @SerializedName("updated_at")
    val updatedAt: Date,
    val student: Student
) {
    data class Student(
        val id: Int,
        val name: String,
        val nim: String,
        @SerializedName("profile_photo")
        val profilePhoto: String?
    )

    enum class Status {
        PENDING,
        APPROVED,
        REJECTED
    }
}